src/main/java/org/broadinstitute/dsm/model/PDF/DownloadPDF.java

```diff
@@ -0,0 +1,291 @@
+                    }
+                }
+                else {
+                    throw new RuntimeException("byte[] was null");
+                }
+            }
+            else {
+                throw new RuntimeException("DDPInstance of participant " + ddpParticipantId + " not found");
+            }
+            response.status(200);
+        }
+        else {
+            response.status(500);
+            throw new RuntimeException("Error missing ddpParticipantId");
+        }
+        return response;
+
+    }
+
+    private byte[] generateSinglePDF(@NonNull String requestBody, String configName, UserDto user, DDPInstance ddpInstance) {
+        byte[] pdfByte = null;
+        if (COVER.equals(configName)) {
+            pdfByte = new MRCoverPDF(this).getMRCoverPDF(requestBody, ddpInstance, user);
+        }
+        else if (IRB.equals(configName)) {
+            String groupId = DDPInstance.getDDPGroupId(ddpInstance.getName());
+            pdfByte = PDFProcessor.getTemplateFromGoogleBucket(groupId + "_IRB_Letter.pdf");
+        }
+        else if (TISSUE.equals(configName) || REQUEST.equals(configName)) {
+            TissueCoverPDF tissueCoverPDF = new TissueCoverPDF(this);
+            pdfByte = tissueCoverPDF.getTissueCoverPDF(ddpInstance, user);
+        }
+        else if (configName != null) {
+            pdfByte = requestPDF(ddpInstance, ddpParticipantId, configName);
+        }
+        return pdfByte;
+    }
+
+    private byte[] getPDFBundle(DDPInstance ddpInstance, String requestBody, UserDto user) {
+        if (ddpInstance != null && StringUtils.isNotBlank(ddpParticipantId)) {
+            ByteArrayOutputStream output = new ByteArrayOutputStream();
+            try {
+                PDFMergerUtility pdfMerger = new PDFMergerUtility();
+                pdfMerger.setDestinationStream(output);
+                pdfs.sort(Comparator.comparing(org.broadinstitute.dsm.model.ddp.PDF::getOrder));
+
+                //make cover pdf first
+                if (pdfs != null && pdfs.size() > 0) {
+                    pdfs.forEach(pdf -> {
+                                byte[] pdfByte;
+                                if (pdf.getOrder() > 0) {
+                                    pdfByte = generateSinglePDF(requestBody, pdf.getConfigName(), user, ddpInstance);
+                                    pdfMerger.addSource(new ByteArrayInputStream(pdfByte));
+                                }
+                            }
+                    );
+                }
+                pdfMerger.mergeDocuments(MemoryUsageSetting.setupTempFileOnly());
+                //todo get page count and add them to the cover/request pdf
+            }
+            catch (IOException e) {
+                throw new FileProcessingException("Unable to merge documents ", e);
+            }
+            return output.toByteArray();
+        }
+        return null;
```

Comment:
I would like to know why you always prefer returning null than returning `Optional`.

Reply:
Honestly I don't see that big of a difference in the usage, both throw an exception when null and both need and if else check. 
However I changed this so the whole code will converge in style

Reply:
@pegahtah, first class is **without optional**, 
there are two examples to work with null:
`getInsuranceNameExample1` - example 1, used widely in legacy projects
`getInsuranceNameExample2` - `guard condition` approach, more or less good, better than first example, but drawback of this approach is that we have many return points which is not good for debugging and etc.
```
public class WithoutOptional {

	
	
	public static void main(String[] args) {
		
		
		Person p = new Person();
		
		String name = getInsuranceNameExample1(p);
		
		String name2 = getInsuranceNameExample2(p);
		
	}

	private static String getInsuranceNameExample2(Person p) {
		Objects.requireNonNull(p);
		Car car = p.getCar();
		if (car == null) return "Unknown";
		Insurance ins = car.getInsurance();
		if (ins == null) return "Unknown";
		String insName = ins.getName();
		if (insName == null) return "Unkown";
		return insName;
	}

	private static String getInsuranceNameExample1(Person p) {
		Objects.requireNonNull(p);
		Car car = p.getCar();
		if (car != null) {
			Insurance ins = car.getInsurance();
			if (ins != null) {
				String insName = ins.getName();
				if (insName != null) {
					return insName;
				}
			}
		}
		return "Unknown";
	}
	
}


class Person {
	private Car car;
	
	public Car getCar() {
		return car;
	}
	
	public void setCar(Car car) {
		this.car = car;
	}
	
}

class Insurance {
	private String name;
	
	public Insurance(String name) {
		this.name = name;
	}
	
	public String getName() {
		return name;
	}
}


class Car {
	private Insurance insurance;
	
	public Insurance getInsurance() {
		return insurance;
	}
	
	public void setInsurance(Insurance insurance) {
		this.insurance = insurance;
	}
}
```
second class, **with optional**:
`getInsuranceName` method to compare `getInsuranceNameExample1` and `getInsuranceNameExample2`, both do the same.
```
public class WithOptional {

	
	
	public static void main(String[] args) {
		
		
		Insurance tbc = new Insurance();
		tbc.setName("TBC");
		
		Car car = new Car();
		car.setInsurance(tbc);
		
		Person p = new Person();
		
		p.setCar(car);
		
		String name = getInsuranceName(p);
		
		System.out.println(name);
		
		
	}

	private static String getInsuranceName(Person p) {
		Objects.requireNonNull(p);
		return p.getCar()
			.flatMap(car -> car.getInsurance())
			.flatMap(ins -> ins.getName())
			.orElse("Unknown");		
	}
	
}


class Person {

	
	private Car car;
	
	public Optional<Car> getCar() {
		return Optional.ofNullable(car);
	}
	
	public void setCar(Car car) {
		this.car = car;
	}
	
	
}

class Car {

	
	private Insurance insurance;
	
	public Optional<Insurance> getInsurance() {
		return Optional.ofNullable(insurance);
	}
	
	public void setInsurance(Insurance ins) {
		this.insurance = ins;
	}
	
}



class Insurance {

	
	private String name;
	
	public Optional<String> getName() {
		return Optional.ofNullable(name);
	}
	
	public void setName(String name) {
		this.name = name;
	}
	
}
```